sdk/test/metrics/histogram_test.cc

```diff
@@ -0,0 +1,278 @@
+    {
+      for (const MetricData &md : smd.metric_data_)
+      {
+        for (const PointDataAttributes &dp : md.point_data_attr_)
+        {
+          actuals.push_back(opentelemetry::nostd::get<HistogramPointData>(dp.point_data));
+        }
+      }
+    }
+    return true;
+  });
+
+  ASSERT_EQ(1, actuals.size());
+
+  const auto &actual = actuals.at(0);
+  ASSERT_EQ(1000275, opentelemetry::nostd::get<int64_t>(actual.sum_));
+  ASSERT_EQ(11, actual.count_);
+  ASSERT_EQ(5, opentelemetry::nostd::get<int64_t>(actual.min_));
+  ASSERT_EQ(1000000, opentelemetry::nostd::get<int64_t>(actual.max_));
+  ASSERT_EQ(
+      std::list<double>({0, 5, 10, 25, 50, 75, 100, 250, 500, 750, 1000, 2500, 5000, 7500, 10000}),
+      actual.boundaries_);
+  ASSERT_EQ(std::vector<uint64_t>({0, 1, 1, 3, 5, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1}),
+            actual.counts_);
+}
+
+TEST(Histogram, UInt64CustomBuckets)
+{
+  MeterProvider mp;
+  auto m = mp.GetMeter("meter1", "version1", "schema1");
+
+  std::unique_ptr<MockMetricExporter> exporter(new MockMetricExporter());
+  std::shared_ptr<MetricReader> reader{new MockMetricReader(std::move(exporter))};
+  mp.AddMetricReader(reader);
+
+  std::shared_ptr<HistogramAggregationConfig> config(new HistogramAggregationConfig());
+  config->boundaries_ = {10, 20, 30, 40};
+  std::unique_ptr<View> view{
+      new View("view1", "view1_description", AggregationType::kHistogram, config)};
+  std::unique_ptr<InstrumentSelector> instrument_selector{
+      new InstrumentSelector(InstrumentType::kHistogram, "histogram1")};
+  std::unique_ptr<MeterSelector> meter_selector{new MeterSelector("meter1", "version1", "schema1")};
+  mp.AddView(std::move(instrument_selector), std::move(meter_selector), std::move(view));
+
+  auto h = m->CreateUInt64Histogram("histogram1", "histogram1_description", "histogram1_unit");
+
+  h->Record(5, {});
+  h->Record(10, {});
+  h->Record(15, {});
+  h->Record(20, {});
+  h->Record(25, {});
+  h->Record(30, {});
+  h->Record(35, {});
+  h->Record(40, {});
+  h->Record(45, {});
+  h->Record(50, {});
+
+  std::vector<HistogramPointData> actuals;
+  reader->Collect([&](ResourceMetrics &rm) {
+    for (const ScopeMetrics &smd : rm.scope_metric_data_)
+    {
+      for (const MetricData &md : smd.metric_data_)
+      {
+        for (const PointDataAttributes &dp : md.point_data_attr_)
+        {
+          actuals.push_back(opentelemetry::nostd::get<HistogramPointData>(dp.point_data));
+        }
+      }
+    }
+    return true;
+  });
+
+  ASSERT_EQ(1, actuals.size());
+
+  const auto &actual = actuals.at(0);
+  ASSERT_EQ(275, opentelemetry::nostd::get<int64_t>(actual.sum_));
+  ASSERT_EQ(10, actual.count_);
+  ASSERT_EQ(5, opentelemetry::nostd::get<int64_t>(actual.min_));
+  ASSERT_EQ(50, opentelemetry::nostd::get<int64_t>(actual.max_));
+  ASSERT_EQ(std::list<double>({10, 20, 30, 40}), actual.boundaries_);
+  ASSERT_EQ(std::vector<uint64_t>({2, 2, 2, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}),
```

Comment:
Shouldn't this be `{2, 2, 2, 2, 2}`?

Reply:
Indeed it should be. Have fixed the code for that. Thanks for pointing out.